speedy-recovery-client/src/components/secured/core/SecuredMainPage.js

```diff
@@ -85,10 +85,17 @@ class SecuredMainPage extends Component {
   }
 
   updateStateAppointments = () => {
-    // TODO: Query SMART, display appointment data for current user
-    const appointments = fhirExampleApp.map((app) => fhirMapAppointment(app, this.props.fhirVersion));
+    if(this.props.user !== null){
+      FhirDataQueryingService.getUserAppointment(this.props.user.id)
+        .then(appointment => {
+          const appointments = this.handleAppointment(appointment);
+          this.setState({ appointments });
+        }).catch(errorMessage=>{this.setState([])});
+    }
+  };
 
-    this.setState({ appointments });
+  handleAppointment = appointment => {
```

Comment:
The unpacking of the nested data should semantically be in the `Querying Service`, also you should make sure the fields actually exists and if not, `console.error()` Therefore I suggest to:
1. Extract the `data.entry.resource` within your `QueryingService` (might want to create a helper method to do that and also do some error handling within that method + write a test for it)
2. In this case, don't use a helper method, but in line 91, directly `appointmentResource => fhirMapAppointment(appointmentResource, this.props.fhirVersion)`